Applications/VGG/jni/main.cpp

```diff
@@ -87,6 +88,12 @@ int main(int argc, char *argv[]) {
     exit(-1);
   }
 
+#ifdef PROFILE
+  auto listener =
+    std::make_shared<nntrainer::profile::GenericProfileListener>();
+  PROFILE_BEGIN(listener);
```

Comment:
I'm not sure,
But removing PROFILE definition check would be better.
We only need to waste making used shared object In non-using PROFILE case, and get clean code benefit.

or,,,

```
std::shard_ptr<nntrainer::profile::ProfileListener> listener;

#ifdef PROFILE
listener = std::make_shared<nntrainer::profile::GenericProfileListener>();
#endif
```

Reply:
done. Thanks.